assets/js/components/DataBlock.js

```diff
@@ -36,193 +36,164 @@ import SourceLink from './SourceLink';
-			source,
-			sparkline,
+	};
+
+	// The `sparkline` prop is passed as a component, but if `invertChangeColor`
+	// is set, we should pass that to `<Sparkline>`. In that case, we clone
+	// the element and add the prop.
+	let sparklineComponent = sparkline;
+	if ( sparklineComponent && invertChangeColor ) {
+		sparklineComponent = cloneElement( sparkline, {
 			invertChangeColor,
-			gatheringData,
-			gatheringDataNoticeStyle,
-		} = this.props;
+		} );
+	}
 
-		const role = 'button' === context ? 'button' : '';
+	let changeFormatted = change;
 
-		// The `sparkline` prop is passed as a component, but if `invertChangeColor`
-		// is set, we should pass that to `<Sparkline>`. In that case, we clone
-		// the element and add the prop.
-		let sparklineComponent = sparkline;
-		if ( sparklineComponent && invertChangeColor ) {
-			sparklineComponent = cloneElement( sparkline, {
-				invertChangeColor,
+	// If changeDataUnit is given, try using it as currency first, otherwise add it as suffix.
+	if ( changeDataUnit ) {
+		if ( changeDataUnit === '%' ) {
+			// Format percentage change with only 1 digit instead of the usual 2.
+			changeFormatted = numFmt( change, {
+				style: 'percent',
+				signDisplay: 'never',
+				maximumFractionDigits: 1,
 			} );
+		} else {
+			changeFormatted = numFmt( change, changeDataUnit );
 		}
+	}
 
-		let changeFormatted = change;
-
-		// If changeDataUnit is given, try using it as currency first, otherwise add it as suffix.
-		if ( changeDataUnit ) {
-			if ( changeDataUnit === '%' ) {
-				// Format percentage change with only 1 digit instead of the usual 2.
-				changeFormatted = numFmt( change, {
-					style: 'percent',
-					signDisplay: 'never',
-					maximumFractionDigits: 1,
-				} );
-			} else {
-				changeFormatted = numFmt( change, changeDataUnit );
-			}
-		}
-
-		// If period is given (requires %s placeholder), add it.
-		if ( period ) {
-			changeFormatted = sprintf( period, changeFormatted );
-		}
-
-		const datapointFormatted =
-			datapoint === undefined
-				? datapoint
-				: numFmt( datapoint, datapointUnit || undefined );
+	// If period is given (requires %s placeholder), add it.
+	if ( period ) {
+		changeFormatted = sprintf( period, changeFormatted );
+	}
 
-		return (
-			<div
-				className={ classnames(
-					'googlesitekit-data-block',
-					className,
-					`googlesitekit-data-block--${ context }`,
-					{
-						'googlesitekit-data-block--selected': selected,
-						'googlesitekit-data-block--is-gathering-data':
-							gatheringData,
-					}
-				) }
-				tabIndex={
-					'button' === context && ! gatheringData ? '0' : '-1'
-				}
-				role={ handleStatSelection && role }
-				onClick={
-					gatheringData
-						? undefined
-						: handleStatSelection && this.handleClick
-				}
-				onKeyPress={
-					gatheringData
-						? undefined
-						: handleStatSelection && this.handleKeyPress
+	const datapointFormatted = datapoint && numFmt( datapoint, datapointUnit );
```

Comment:
Same here, I don't know for sure but explicitly checking whether the `datapoint` variable is `undefined` might have some sense that will be lost after your changes. Please, revert it as it was before.